tests/XrdHttpTests/XrdHttpTests.cc

```diff
@@ -606,4 +606,50 @@ TEST(XrdHttpTests,encodeOpaqueTest) {
   for(auto [decoded,encoded]: decodedEncodedOpaque) {
     ASSERT_EQ(encoded,encode_opaque(decoded));
   }
-}
\ No newline at end of file
+}
+
+TEST(XrdHttpTests, hasCGIParamAuthz) {
```

Comment:
@ccaffy Added the requested unit test for checking against various conditions, including the presence of false positives e.g. `authauthz`